### PR TITLE
Prepare for Maven Publishing

### PR DESCRIPTION
**Pull request by lucagiovagnoli**
This is required by Sonatype to publish to Maven Central:
https://issues.sonatype.org/browse/OSSRH-51286

Helps fixing #19

### CODE REVIEW

**Comment by lucagiovagnoli** [comment]
@cortinico to have a look at Maven Central publishing.

File: build.sbt [hunk 1]
```diff
@@ -101,12 +101,20 @@ lazy val root = Project("spark-redshift", file("."))
      * Release settings *
      ********************/
 
+    publishTo := {
+      val nexus = "https://oss.sonatype.org/"
```
Comment (cortinico) [inline build.sbt]: Please note that you need to specify also Nexus username and password.
Those are completely unrelated with the GPG credentials.

Comment (lucagiovagnoli) [reply build.sbt]: Oh yes I put those in a file offline on disk (not in the repo).
I created sonatype.sbt following https://www.scala-sbt.org/0.13/docs/Using-Sonatype.html
and I also created a .sbt/.credentials file following: https://www.scala-sbt.org/1.x/docs/Publishing.html#Credentials

see line https://github.com/spark-redshift-community/spark-redshift/pull/28/files#diff-fdc3abdfd754eeb24090dbd90aeec2ceR60

I hope that's enough?

**Comment by lucagiovagnoli** [comment]
@lydian @talalryz @tinaxi would you mind having a look at this? I need to try a few things out to publish this library out to maven. I'll probably require a few more iterations.

File: build.sbt [hunk 2]
```diff
@@ -101,12 +101,20 @@ lazy val root = Project("spark-redshift", file("."))
...
+        Some("releases"  at nexus + "service/local/staging/deploy/maven2")
+    },
+
     publishMavenStyle := true,
     releaseCrossBuild := true,
     licenses += ("Apache-2.0", url("http://www.apache.org/licenses/LICENSE-2.0")),
     releasePublishArtifactsAction := PgpKeys.publishSigned.value,
 
-    pomExtra :=
+      pomExtra :=
```
Comment (tinaxi) [inline build.sbt]: Nit: extra whitespaces here.

**Comment by tinaxi** [comment]
LGTM

**Comment by lucagiovagnoli** [comment]
This worked! https://oss.sonatype.org/content/repositories/snapshots/io/github/spark-redshift-community/spark-redshift_2.11/4.0.0-SNAPSHOT/
